partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/PartitionerConfigServiceRequestValidator.java

```diff
@@ -0,0 +1,90 @@
+package org.hypertrace.partitioner.config.service;
+
+import io.grpc.Status;
+import java.util.HashMap;
+import java.util.Map;
+import org.hypertrace.partitioner.config.service.v1.*;
+
+public class PartitionerConfigServiceRequestValidator {
+
+  public void validateOrThrow(GetPartitionerProfileRequest request) {
+    if (request.getProfileName().isBlank()) {
+      throw Status.INVALID_ARGUMENT.withDescription("profile cannot be empty").asRuntimeException();
+    }
+  }
+
+  public void validateOrThrow(PutPartitionerProfilesRequest request) {
+    request
+        .getProfilesList()
+        .forEach(
+            profile -> {
+              if (profile.getName().isBlank()) {
+                throw Status.INVALID_ARGUMENT
+                    .withDescription("profile cannot be empty")
+                    .asRuntimeException();
+              }
+              if (profile.getGroupsCount() == 0) {
+                throw Status.INVALID_ARGUMENT
+                    .withDescription("partition_groups cannot be empty")
+                    .asRuntimeException();
+              }
+              if (profile.getPartitionKey().isBlank()) {
+                throw Status.INVALID_ARGUMENT
+                    .withDescription("partition key cannot be empty")
+                    .asRuntimeException();
+              }
+              profile
+                  .getGroupsList()
```

Comment:
at least one group

Reply:
already checking for groupsCount == 0 in L30 above